### DISCUSSION TITLE
Configure ODoH relay and server

### DISCUSSION THREAD

**Discussion by AmirSalari** [DNSCrypt/dnscrypt-proxy#2203] [Q&A]
I tried to configure the dnscrypt-proxy to use my own ODoH relay and server by adding their DNS stamp as static variables at the end of "dnscrypt-proxy.toml" file like:

```
[static]

  [static.myProxy]
     stamp = 'sdns://hQcAAAAAAAAAEjE2NC42Ny4xMjYuMzk6ODEzMQAABi9wcm94eQ'

  [static.myServer]
     stamp = 'sdns://BQcAAAAAAAAAEjE2NC42Ny4xMjYuNDA6ODEzMQUvb2RvaA'
```

and then addressing them in "routes" variable ( in [anonymized_dns] section in "dnscrypt-proxy.toml" file), like this:

```
 routes = [
    { server_name='myServer', via=['myProxy'] }
 ]
```

but I'm getting this error when I run "dnscrypt-proxy.exe":
```
[ERROR] No valid network configuration for [myServer]
[NOTICE] dnscrypt-proxy is waiting for at least one server to be reachable
```

**Comment by lifenjoiner** [comment]
It means the line you specified is not validated succeeded.
Maybe your servers are not configured correct.

**Reply by AmirSalari** [reply]
Thanks for the response, how should I configure my server correctly? this is my relay IP: 164.67.126.39 and this is my server IP 164.67.126.40. In the DNS stamp, I set the path for the server to "/dns-query" and for the relay, I set it to "/proxy". I implemented the server and proxy based on the ODoH GitHub page: https://github.com/cloudflare/odoh-server-go .

Not sure what is the problem with my server and relay configuration.

**Reply by lifenjoiner** [reply]
I'm not familiar with those servers either. Please follow their docs.

**Comment by jedisct1** [comment]
`odoh-server-go` was a proof concept. The project is not maintained any more and was never meant to run actual servers. It's actually not safe to expose it to the internet.

In order to run an ODoH server, you should run [doh-server](https://github.com/dnscrypt/doh-server).

For ODoH relays, well... there's nothing yet. Operating an ODoH relay essentially means running an open HTTP proxy, which has security implications. The specification itself still requires some work regarding security considerations.

Still, I'm going to opensource the relay code used by `odohrelay-crypto-sx` soon. It currently depends on Fastly's proprietary APIs, so I need to make it more portable first.
